### PR TITLE
Fix activation bug for GRU operator

### PR DESCRIPTION
**Pull request by yihuaxu**
It will can not use the AVX intrinsic function without 8-bytes aligned issue when used the GRU operator.
Just re-implement the relevant functions to fix the issue.

Platform: Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz
Model Path: models/fluid/chinese_ner
Batch Size: 6
Command: python infer.py --device CPU --profile
Data Source: Use original date provided in the model Chinese_NER in the github.

The following is the comparison with the different scenarios.

image

### CODE REVIEW

File: paddle/fluid/operators/math/detail/gru_cpu_kernel.h [hunk 1]
```diff
@@ -89,22 +89,42 @@ void hl_avx_gru_forward_reset_output(OpResetOutput op_reset_output,
...
+                     r_value_update_gate);
+    _mm256_storeu_ps(reinterpret_cast<float *>(reset_gate + i_offset),
+                     r_value_reset_gate);
+    _mm256_storeu_ps(reinterpret_cast<float *>(reset_output_value + i_offset),
+                     r_value_reset_output);
+
+    i_offset += step_size;
+    if (i == steps - 1) {
+      if (remain > 0) {
+        i_offset += last_offset;
```
Comment (tensor-tang) [inline paddle/fluid/operators/math/detail/gru_cpu_kernel.h]: 这里用remain的方式会不会有问题，重复计算的那一部分会不会导致结果有误，特别是有exp的那一部分。

请在https://github.com/PaddlePaddle/Paddle/blob/f15def64034b800b5e9bec5f27525a497a7d8210/python/paddle/fluid/tests/unittests/test_gru_op.py#L102-L106 添加对应的单侧，并保证能跑到现在添加的这一段代码。现在的单侧dtype是double的，frame size是5并不会走进这里。

Comment (yihuaxu) [reply paddle/fluid/operators/math/detail/gru_cpu_kernel.h]: 刚修改了下，并加了针对float32的unitest
https://github.com/PaddlePaddle/Paddle/pull/13275/commits/42584ec8249ca75ac78dbdd6329c24a7261f8ab6

File: python/paddle/fluid/tests/unittests/test_gru_op.py [hunk 2]
```diff
@@ -153,6 +154,55 @@ def test_check_grad(self):
         self.check_grad(['Input', 'H0', 'Weight', 'Bias'], ['Hidden'])
 
 
+class TestGRUOp2(TestGRUOp):
+    def setUp(self):
+        self.op_type = "gru"
+        self.lod = [[2, 4, 3]]
+        self.D = 19
+        self.is_reverse = False
```
Comment (tensor-tang) [inline python/paddle/fluid/tests/unittests/test_gru_op.py]: 这段往下是重复的吧？
 应该可以仿照https://github.com/PaddlePaddle/Paddle/pull/13275/commits/42584ec8249ca75ac78dbdd6329c24a7261f8ab6#diff-cb83a83565f99e04f3e6ba4bb470982eR206 这的做法，

加dtype和增加self.D的设置就好了。

**Comment by tensor-tang** [comment]
Thanks @yihuaxu , since the implementation has changed, how about the performance now?

**Comment by yihuaxu** [comment]
Just update the latest profile data with this implementation

**Comment by luotao1** [comment]
Please pass the ci.
```
[02:04:53][Step 1/1] 	189 - test_fusion_gru_op (Failed)
```

**Comment by yihuaxu** [comment]
> Please pass the ci.
> 
> ```
> [02:04:53][Step 1/1] 	189 - test_fusion_gru_op (Failed)
> ```

How to verify it about CI? just modify the gru parameters and it can pass the test case for "test_fusion_gru_op.py" .

**Comment by luotao1** [comment]
`ctest -R test_fusion_gru_op -V`

**Comment by yihuaxu** [comment]
> `ctest -R test_fusion_gru_op -V`

It can pass with the latest commit. 
https://github.com/PaddlePaddle/Paddle/pull/13275/commits/04542328fa5dfe45988788c1444fc03885ff0f7e

File: python/paddle/fluid/tests/unittests/test_gru_op.py [hunk 3]
```diff
@@ -31,7 +31,7 @@ def gru(
         is_reverse,
         act_state,
         act_gate,
-        dtype):
+        dtype='float64'):
```
Comment (tensor-tang) [inline python/paddle/fluid/tests/unittests/test_gru_op.py]: default should be float32

Comment (yihuaxu) [reply python/paddle/fluid/tests/unittests/test_gru_op.py]: Done. According to the old rule, it should be "float64". anyway, just change the default type as 'float32'.

**Review by tensor-tang — APPROVED** [review]
Thanks @yihuaxu

**Pull request merged**